### PR TITLE
[SofaConstraint] Sanitize

### PR DESCRIPTION
**Pull request by alxbilger**
Cleaning!!!

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: modules/SofaConstraint/src/SofaConstraint/LCPConstraintSolver.cpp [hunk 1]
```diff
@@ -469,7 +469,7 @@ void LCPConstraintSolver::MultigridConstraintsMerge_Compliance()
 void LCPConstraintSolver::MultigridConstraintsMerge_Spatial()
 {
     const int merge_spatial_step = this->merge_spatial_step.getValue();
-    const int merge_spatial_shift = 0; // merge_spatial_step/2
+    constexpr int merge_spatial_shift = 0; // merge_spatial_step/2
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/LCPConstraintSolver.cpp:472]: 🤔 should we not just remove this variable ?

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: modules/SofaConstraint/src/SofaConstraint/LCPConstraintSolver.cpp (hunk 1 above)
Comment (alxbilger) [reply modules/SofaConstraint/src/SofaConstraint/LCPConstraintSolver.cpp:472]: It's an option, but from the comment, we can see that an intention is to set this variable to `merge_spatial_step/2`.
Based on the poor code quality in this module, I would keep it. It's not like it is the only code quality concern in this module. There are many other things to clean in this module.

**Comment by fredroy** [comment]
2 scenes are crashing:
 - examples/Components/mapping.DeformableOnRigidFrameMappingConstraints.scn
 - examples/Components/animationloop.FreeMotionAnimationLoop.scn

Seens to be related to PrecomputedConstraintCorrection (which is modified in this PR)

File: modules/SofaConstraint/src/SofaConstraint/PrecomputedConstraintCorrection.inl [hunk 2]
```diff
@@ -739,7 +734,7 @@ void PrecomputedConstraintCorrection<DataTypes>::applyContactForce(const lineara
                 {
                     DXbuf += appCompliance[ offset2 + i ] * Fbuf[i];
                 }
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/PrecomputedConstraintCorrection.inl:736]: I guess the problem is here (i is used in two loops)

Comment (alxbilger) [reply modules/SofaConstraint/src/SofaConstraint/PrecomputedConstraintCorrection.inl:736]: Thanks for pointing to this direction

**Comment by fredroy** [comment]
👍

**Review by fredroy — APPROVED** [review]

**Pull request merged**